### PR TITLE
slice: add SliceCompare function

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
add `SliceCompare` and `SliceDedupe` function

### CODE REVIEW

File: slice.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package gocommon
+
+import "golang.org/x/exp/slices"
+
+func SliceCompare[T comparable](x, y []T) bool {
```
Comment (bk201) [inline slice.go:5]: Can we add some documents for both functions?

Comment (Vicente-Cheng) [reply slice.go:6]: Thanks for the reminder! Updated!

File: slice_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	c := []string{"a", "b", "c", "d"}
+	d := []string{"a", "b"}
+	e := []string{}
+	f := []string{}
+
+	s.Equal(true, SliceCompare(a, b), "SliceCompare should return true")
+	s.Equal(false, SliceCompare(a, c), "SliceCompare should return false")
+	s.Equal(false, SliceCompare(c, d), "SliceCompare should return false")
+	s.Equal(true, SliceCompare(e, f), "SliceCompare should return true")
+}
```
Comment (FrankYang0529) [inline slice_test.go:48]: This function can't handle following case:

```go
	g := []int{1, 1, 2}
	h := []int{1, 2, 2}
	s.Equal(false, SliceCompare(g, h), "SliceCompare should return false")
```

**Review by FrankYang0529 — CHANGES_REQUESTED** [review]

File: slice_test.go (hunk 2 above)
Comment (Vicente-Cheng) [reply slice_test.go:54]: Nice catch! Updated!

File: slice.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			if xItem == yItem {
+				y = slices.Delete(y, id, id+1)
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
```
Comment (FrankYang0529) [inline slice.go:26]: In golang, the slice header contains a pointer to the start of the array. If we modify input, we also update user's content. It's better to copy it first.

```go
    yCpy := make([]T, len(y))
	copy(yCpy, y)
	for _, xItem := range x {
		found := false
		for id, yItem := range yCpy {
			if xItem == yItem {
				yCpy = slices.Delete(yCpy, id, id+1)
				found = true
				break
			}
		}
		if !found {
			return false
		}
	}
```

File: slice_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	e := []int{}
+	f := []int{}
+	g := []int{1, 1, 2}
+	h := []int{1, 2, 2}
+
+	s.Equal(true, SliceCompare(a, b), "SliceCompare should return true")
+	s.Equal(false, SliceCompare(a, c), "SliceCompare should return false")
+	s.Equal(false, SliceCompare(c, d), "SliceCompare should return false")
+	s.Equal(true, SliceCompare(e, f), "SliceCompare should return true")
+	s.Equal(false, SliceCompare(g, h), "SliceCompare should return false")
```
Comment (FrankYang0529) [inline slice_test.go:36]: Also, we can test whether input is updated here.

```go
	gCpy := []int{1, 1, 2}
	hCpy := []int{1, 2, 2}

    // ...
    s.Equal(false, SliceCompare(g, h), "SliceCompare should return false")
	s.Equal(true, reflect.DeepEqual(g, gCpy), "SliceCompare should not update input")
	s.Equal(true, reflect.DeepEqual(h, hCpy), "SliceCompare should not update input")
```

**Review by FrankYang0529 — CHANGES_REQUESTED** [review]

File: slice.go (hunk 3 above)
Comment (Vicente-Cheng) [reply slice.go:28]: Updated! Thanks for the review!

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks!

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks.

**Review by bk201 — APPROVED** [review]

**Pull request merged**